From c2dbfdf9d09642c8ee24c2c9377b3f73f0e2ad0e Mon Sep 17 00:00:00 2001
From: Shivani Agarwal <shivania2@vmware.com>
Date: Fri, 9 Sep 2022 16:26:27 +0000
Subject: [PATCH 26/29] Fix etcd denials for K8's deployment with containerd

Changes are added to resolve following etcd denials

avc:  denied  { add_name } for  pid=10891 comm="etcd" name=".touch"
scontext=system_u:system_r:container_t:s0:c506,c833
tcontext=system_u:object_r:var_lib_t:s0 tclass=dir permissive=1
---
 policy/modules/contrib/container.if | 78 +++++++++++++++++++++++++++++
 policy/modules/contrib/container.te |  4 ++
 2 files changed, 82 insertions(+)

diff --git a/policy/modules/contrib/container.if b/policy/modules/contrib/container.if
index 6e10d23..dacc0da 100644
--- a/policy/modules/contrib/container.if
+++ b/policy/modules/contrib/container.if
@@ -252,6 +252,84 @@ interface(`container_runtime_watch_bin_files',`
          allow $1 bin_t:file watch_file_perms;
 ')
 
+########################################
+## <summary>
+##      Manage container var lib files.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`container_runtime_manage_var_lib_files',`
+        gen_require(`
+                type var_lib_t;
+        ')
+
+        files_search_locks($1)
+        files_search_var_lib($1)
+        manage_files_pattern($1, var_lib_t, var_lib_t)
+        manage_lnk_files_pattern($1, var_lib_t, var_lib_t)
+')
+
+########################################
+## <summary>
+##      Rename var lib files
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`container_runtime_rename_var_lib_files',`
+        gen_require(`
+                type var_lib_t;
+        ')
+
+        rename_dirs_pattern($1, var_lib_t, var_lib_t)
+        rename_files_pattern($1, var_lib_t, var_lib_t)
+')
+
+########################################
+## <summary>
+##      Create var lib files.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+## <rolecap/>
+#
+interface(`container_runtime_create_var_lib_files',`
+        gen_require(`
+                type var_lib_t;
+        ')
+
+        allow $1 var_lib_t:dir create;
+        allow $1 var_lib_t:file create;
+')
+
+########################################
+## <summary>
+##      Map var lib files.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>+## </param>
+## <rolecap/>
+#
+interface(`container_runtime_map_var_lib_files',`
+        gen_require(`
+                type var_lib_t;
+        ')
+
+        allow $1 var_lib_t:file map;
+')
+
 ########################################
 ## <summary>
 ##	Manage container share files.
diff --git a/policy/modules/contrib/container.te b/policy/modules/contrib/container.te
index 6bbb84e..1ac42d1 100644
--- a/policy/modules/contrib/container.te
+++ b/policy/modules/contrib/container.te
@@ -1302,3 +1302,7 @@ term_mount_pty_fs(container_engine_t)
 container_runtime_read_kubernetes_files(container_t)
 container_runtime_watch_kubernetes_files(container_t)
 container_runtime_watch_bin_files(container_t)
+container_runtime_manage_var_lib_files(container_t)
+container_runtime_rename_var_lib_files(container_t)
+container_runtime_create_var_lib_files(container_t)
+container_runtime_map_var_lib_files(container_t)
-- 
2.45.1

